backend/src/test/java/com/redhat/cloud/notifications/db/BehaviorGroupResourcesTest.java

```diff
@@ -48,6 +49,39 @@
     @Inject
     BehaviorGroupResources behaviorGroupResources;
 
+    @Test
+    void shouldThrowExceptionWhenBehaviorGroupWithAccountIdAndDisplayNameAlreadyExistsWhenUpdating() {
+        final BadRequestException exception = assertThrows(BadRequestException.class, () -> {
+            Bundle bundle = resourceHelpers.createBundle();
+            final BehaviorGroup behaviorGroup = resourceHelpers.createBehaviorGroup(DEFAULT_ACCOUNT_ID, "displayName", bundle.getId());
+            resourceHelpers.updateBehaviorGroup(behaviorGroup);
```

Comment:
I was about to approve and then that test made me realize there's a problem: updating `behaviorGroupA` with the display name of `behaviorGroupB` should be forbidden (this works) but updating `behaviorGroupA` without changing its display name should not be forbidden (this doesn't work).

For the update only, the SQL query needs to be something like:
```
SELECT COUNT(*) FROM BehaviorGroup b WHERE b.displayName = :displayName and b.accountId IS NULL AND id != :id
```

Reply:
This is still a thing, isn't it? @bennetelli